ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/ScalablePushRegistry.java

```diff
@@ -364,4 +457,129 @@ private void runLatest(final LatestConsumer latestConsumerToRun) {
   private String getLatestConsumerGroupId() {
     return sourceApplicationId + LATEST_CONSUMER_GROUP_SUFFIX;
   }
+
+  private synchronized void catchupAssignmentUpdater(
+      final Collection<TopicPartition> newAssignment
+  ) {
+    for (CatchupConsumer catchupConsumer : catchupConsumers.values()) {
+      catchupConsumer.newAssignment(newAssignment);
+    }
+  }
+
+  /**
+   * Creates the latest consumer and its underlying kafka consumer.
+   * @param processingQueue The queue on which to send an error if anything goes wrong
+   * @return The new LatestConsumer
+   */
+  private CatchupConsumer createCatchupConsumer(
```

Comment:
nit: wdyt about making all non-syncrhonized methods static and passing in everything we need here to ensure that we're not abusing thread unsafe state?